### PR TITLE
feat: add CREATE SOURCE STREAM syntax and metadata info

### PR DESCRIPTION
**Pull request by jzaralim**
### Description 
The STREAM version of #7945. 

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-rest-app/src/test/resources/ksql-plan-schema/schema.json [hunk 1]
```diff
@@ -1070,6 +1074,9 @@
         "leftJoinColumnName" : {
           "type" : "string"
         },
+        "leftJoinExpression" : {
```
Comment (spena) [inline ksqldb-rest-app/src/test/resources/ksql-plan-schema/schema.json:1077]: Why did this change? Is this move going to affect the historic plans?

Comment (jzaralim) [reply ksqldb-rest-app/src/test/resources/ksql-plan-schema/schema.json:1077]: I got this by running KsqlPlanSchemaTest.runMeToRegenerateSchemaFile. Perhaps this change is related to another PR?

Comment (spena) [reply ksqldb-rest-app/src/test/resources/ksql-plan-schema/schema.json:1077]: Probably. You can manually reset that change and just leave the isSource field and see what happens. I have done that sometimes when I don't know why the schema or plan has changes I don't know.

Comment (jzaralim) [reply ksqldb-rest-app/src/test/resources/ksql-plan-schema/schema.json]: Reverted

**Review by spena — APPROVED** [review]
LGTM

**Pull request merged**